pkg/platform/runtime/setup/setup.go

```diff
@@ -550,9 +550,21 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 		return nil, nil, errs.Wrap(err, "Failed to compute artifacts to build")
 	}
 
-	// Output a change summary if applicable.
-	if len(oldBuildPlanArtifacts) > 0 {
-		changesummary.New(s.out).ChangeSummary(changedArtifacts, artifactsToBuild, oldBuildPlanArtifacts)
+	// Output a dependency summary if applicable.
+	if !fileutils.DirExists(s.target.Dir()) {
```

Comment:
There's a potential order of operations issue here for future updates. The dir existing has implicit behaviour tied to it, which we may break without realizing in future updates. 

I think at this point we should have access to the trigger, can we identify checkout from there and then run this logic? 

I think either we need to make this logic more explicit, or we need to have a test covering it to ensure we don't break it.

Reply:
Ahhhh, thanks! Checking the trigger is exactly what I wanted. I definitely did not like the directory existing check, but I couldn't figure out how else to figure out if this was a fresh checkout or not.